lib/solveEquation/stepThrough.js

```diff
@@ -58,8 +58,11 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
     steps = addSimplificationSteps(steps, equation, debug);
     if (steps.length > 0) {
       const lastStep = steps[steps.length - 1];
-      equation = Equation.createEquationFromString(
-        lastStep.newEquation.print(), equation.comparator);
+      equation.leftNode = lastStep.newEquation.leftNode.clone();
+      equation.rightNode = lastStep.newEquation.rightNode.clone();
+      // console.log(left);
+      // equation = Equation.createEquationFromString(
+      //   lastStep.newEquation.print(), equation.comparator);
```

Comment:
yes I think cloning is best - good call. glad it makes it faster too!

I think cloning resets changeGroups, but if it doesn't we can manually add that too.

I think it's reasonable to have `print()` not be able to be re-parsed, especially if there's an option to turn off the thing that makes it unable to be re-parsed (ie, you can have it print with `...` and repeating fractions iff you're okay with not being able to re-parse)

can't you serialize/deserialize with JSONifying the struct? or is that not what you meant